protocol/src/main/java/net/md_5/bungee/protocol/packet/ExperimentalFeatures.java

```diff
@@ -0,0 +1,45 @@
+package net.md_5.bungee.protocol.packet;
+
+import io.netty.buffer.ByteBuf;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import net.md_5.bungee.protocol.AbstractPacketHandler;
+import net.md_5.bungee.protocol.DefinedPacket;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class ExperimentalFeatures extends DefinedPacket
+{
+    private List<String> features = new ArrayList<>();
+
+    @Override
+    public void read(ByteBuf buf)
+    {
+        for ( int i = 0; i < readVarInt( buf ); i++ )
```

Comment:
Broken loop

Reply:
> Broken loop

Oh sorry, i decided to refactor the code and completely forgot that readVarInt adds readIndex to the buffer